third_party/amd/CMakeLists.txt

```diff
@@ -1,6 +1,5 @@
 include_directories(${CMAKE_CURRENT_SOURCE_DIR}/include)
 include_directories(${CMAKE_CURRENT_BINARY_DIR}/include)
-add_compile_definitions(USE_ROCM=1)
```

Comment:
I've removed this definition and added it for libraries.
Old way `USE_ROCM` is not visible in objects linked with AMD libraries. (i.e. `USE_ROCM` is not defined for `triton_opt` target)